scripts/generator/graphql_generator/csharp/SDK/BaseLoader.cs.erb

```diff
@@ -0,0 +1,58 @@
+namespace <%= namespace %>.SDK {
+    /// <summary>
+    /// An abstract base that must be defined by classes that will perform network communication.
+    /// </summary>
+    public abstract class BaseLoader {
+        /// <summary>
+        /// Domain of a Shopify store.
+        /// </summary>
+        public string Domain {
+            get {
+                return _Domain;
+            }
+        }
+
+        /// <summary>
+        /// Access token used to communicate with a Shopify store.
+        /// </summary>
+        public string AccessToken {
+            get {
+                return _AccessToken;
+            }
+        }
+
+        private string _AccessToken;
+        private string _Domain;
+
+        public BaseLoader(string domain, string accessToken) {
+            _AccessToken = accessToken;
+            _Domain = domain;
+
+            SetHeader("Content-Type", "application/graphql");
+            SetHeader("X-SDK-Version", "ShopifyBuy.VERSION");
+            SetHeader("X-Shopify-Storefront-Access-Token", accessToken);
+            SetHeader("X-SDK-Variant", SDKVariantName());
```

Comment:
By moving all this here and exposing an abstract method named `SDKVariantName`, we can factor out the required headers for communicating to our GraphQL services away from any SDK-specific loader.